NitroxServer/GameLogic/Entities/EntityData.cs

```diff
@@ -45,5 +45,38 @@ public static EntityData From(List<Entity> entities)
         {
             return new EntityData { Entities = entities };
         }
+
+        public bool Equals(EntityData other)
```

Comment:
I have mixed feelings about generating equals/hashcode for so many classes to support one unit test. These data models are used/indexed in strange ways.  Any type of missed field (now or in the future) will result in strange behavior that is very difficult to diagnose.  They are also very bad on performance as many of them are recursively calling children equality; when in actual production code we only need reference equality. The equals/hashcode methods are also not unit tested, so net coverage is going down - we are essentially adding what is functionally an integration test.  Instead, I would be more comfortable adding individual tests to verify each model can be serialized/deserialized properly.  That way a missed member would only be impacting a test and not actual production code.

Reply:
- I know it's not the nicest way but the only other solution I can think of is a huge chunk of equality checks in `WorldPersistenceTest` which nobody will update either.
- I don't see why there will be a performance impact. The equality checks of the models are just used in the test (and shouldn't be somewhere else because they just store data for serialization).
- Net coverage isn't going down on my side: 
![grafik](https://user-images.githubusercontent.com/23176718/102893534-0cf81200-4462-11eb-8371-81abfc16afc5.png)
- I am fine with a test class for each model but that would result in 12+ test classes. Just tell me if I should do it that way, I am really eager to implement this because in the past serialization broke a few times and a test like this could indicate an error immediately.

Reply:
The behavior and performance changes can occur when indexing any of these datastructures in a dictionary or hashset.  Whenever you insert into these datastructures, it will execute these methods; this happens recursively for children if they are included in the equals/hashcode.  I don't trust our codebase enough to keep all of these updated and think the possibility of inadvertent bugs is likely.  Would rather punt those to the unit tests.

As for the net coverage, all of these lines will show as covered because they get executed from a unit test.  However, there is nothing doing actual assertions that is saying that the equality/hashcode operations are performing as expected.  So they are showing as covered because they are used to check equality as a side-effect of another unit test for serialization.

 I think we can have a middle ground with a single serialization test class with different methods for each object type. If they all collectively serialize/deserialize, then it would be a significant increase in confidence.